tests/integration/targets/meraki_mr_l7_firewall/tasks/tests.yml

```diff
@@ -0,0 +1,511 @@
+          application:
+            name: Advertising
+    register: application_category_rule
+
+  - name: Debug application_category_rule
+    ansible.builtin.debug:
+      var: application_category_rule
+
+  - name: Assert application_category_rule is changed
+    ansible.builtin.assert:
+      that:
+        - application_category_rule is changed
+
+  - name: Create firewall rule for application category with ID and conflict
+    cisco.meraki.meraki_mr_l7_firewall:
+      auth_key: '{{ auth_key }}'
+      org_name: '{{ test_org_name }}'
+      net_id: '{{ net }}'
+      ssid: TestSSID
+      state: present
+      rules:
+        - type: application_category
+          application:
+            id: meraki:layer7/category/27
+    register: application_category_rule_id_conflict
+
+  - name: Assert application_category_rule_id_conflict is not changed
+    ansible.builtin.assert:
+      that:
+        - application_category_rule_id_conflict is not changed
+
+  - name: Create firewall rule for application category with ID
+    cisco.meraki.meraki_mr_l7_firewall:
+      auth_key: '{{ auth_key }}'
+      org_name: '{{ test_org_name }}'
+      net_id: '{{ net }}'
+      state: present
+      ssid: TestSSID
+      rules:
+        - type: application_category
+          application:
+            id: meraki:layer7/category/24
+    register: application_category_rule_id
+
+  - name: Assert application_category_rule_id is changed
+    ansible.builtin.assert:
+      that:
+        - application_category_rule_id is changed
+
+  - name: Create firewall rule for host
+    cisco.meraki.meraki_mr_l7_firewall:
+      auth_key: '{{ auth_key }}'
+      org_name: '{{ test_org_name }}'
+      net_id: '{{ net }}'
+      ssid: TestSSID
+      state: present
+      rules:
+        - type: host
+          host: asdf.com
+    register: host_rule
+
+  - name: Assert host_rule is changed
+    ansible.builtin.assert:
+      that:
+        - host_rule is changed
+
+  - name: Create firewall rule for port
+    cisco.meraki.meraki_mr_l7_firewall:
+      auth_key: '{{ auth_key }}'
+      org_name: '{{ test_org_name }}'
+      net_id: '{{ net }}'
+      state: present
+      ssid: TestSSID
+      rules:
+        - type: port
+          port: 1234
+    register: port_rule
+
+  - name: Assert port_rule is changed
+    ansible.builtin.assert:
+      that:
+        - port_rule is changed
+
+  - name: Create multiple firewall rules
+    cisco.meraki.meraki_mr_l7_firewall:
+      auth_key: '{{ auth_key }}'
+      org_name: '{{ test_org_name }}'
+      net_id: '{{ net }}'
+      ssid: TestSSID
+      state: present
+      rules:
+        - type: application_category
+          application:
+            id: meraki:layer7/category/27
+        - policy: deny
+          type: port
+          port: 8080
+    register: multiple_rules
+
+  - name: Debug multiple_rules
+    ansible.builtin.debug:
+      var: multiple_rules
+
+  - name: Assert multiple_rules.data.rules | length == 3
```

Comment:
This is failing as the length isn't matching. The output I see is below as well.

```
        "data": {
            "rules": [
                {
                    "application_category": {
                        "id": "meraki:layer7/category/27",
                        "name": "Advertising"
                    },
                    "policy": "deny",
                    "type": "applicationCategory"
                },
                {
                    "policy": "deny",
                    "port": "8080",
                    "type": "port"
                }
            ]
        },
```

```
TASK [meraki_mr_l7_firewall : Assert multiple_rules.data.rules | length == 3] ***
fatal: [localhost]: FAILED! => {
    "assertion": "multiple_rules.data.rules | length == 3",
    "changed": false,
    "evaluated_to": false,
    "msg": "Assertion failed"
}
```